onnxruntime/core/providers/coreml/builders/impl/pool_op_builder.cc

```diff
@@ -54,23 +110,50 @@ Status PoolOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
 bool PoolOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& /* initializers */, const Node& node,
                                       const logging::Logger& logger) const {
   const auto& op_type = node.OpType();
-  if (op_type != "GlobalAveragePool" && op_type != "GlobalMaxPool") {
-    LOGS(logger, VERBOSE) << "[" << op_type << "] is not supported";
-    return false;
-  }
+  const auto& input_defs = node.InputDefs();
 
   std::vector<int64_t> input_shape;
-  if (!GetShape(*node.InputDefs()[0], input_shape, logger))
+  if (!GetShape(*input_defs[0], input_shape, logger))
     return false;
 
   const auto input_size = input_shape.size();
   if (input_size != 4) {
     LOGS(logger, VERBOSE)
         << op_type << " only supports rank-4 tensor, input ["
-        << node.InputDefs()[0]->Name() << "] has actual dim count " << input_size;
+        << input_defs[0]->Name() << "] has actual dim count " << input_size;
     return false;
   }
 
+  if (op_type == "AveragePool" || op_type == "MaxPool") {
+    NodeAttrHelper helper(node);
+    const auto storage_order = helper.Get("storage_order", 0);
+    if (storage_order == 1) {
+      LOGS(logger, VERBOSE) << "storage_order == 1 is not supported";
+      return false;
+    }
+
+    if (helper.Get("kernel_shape", std::vector<int32_t>{1, 1}).size() != 2) {
+      LOGS(logger, VERBOSE) << "Only pooling 2d is supported";
+      return false;
+    }
+
+    if (helper.Get("ceil_mode", 0) == 1) {
+      LOGS(logger, VERBOSE) << "ceil_mode == 1 is not supported for pooling";
+      return false;
+    }
+
```

Comment:
Does this work now, or the coreml tools do something different to avoid a ceil_mode of 1?

[https://github.com/onnx/onnx-coreml/issues/573](https://github.com/onnx/onnx-coreml/issues/573)

Reply:
Yes, this can be supported later by adjusting the padding

https://stackoverflow.com/questions/59906456/in-pytorchs-maxpool2d-is-padding-added-depending-on-ceil-mode

https://github.com/apple/coremltools/blob/1931758aae383c83daddfc56f11a24a9d2bf4b87/coremltools/converters/mil/frontend/torch/ops.py#L621-L644